src/java.desktop/macosx/classes/sun/lwawt/macosx/CPrinterJob.java

```diff
@@ -724,11 +734,24 @@ private void setPrinterServiceFromNative(String printerName) {
     }
 
     private Rectangle2D getPageFormatArea(PageFormat page) {
-        Rectangle2D.Double pageFormatArea =
-            new Rectangle2D.Double(page.getImageableX(),
-                    page.getImageableY(),
-                    page.getImageableWidth(),
-                    page.getImageableHeight());
+        return getPageFormatArea(page, 1, 1);
+    }
+
+    private Rectangle2D getPageFormatArea(PageFormat page, double scaleX, double scaleY) {
+        Rectangle2D.Double pageFormatArea;
+        if (scaleX != 1 && scaleY != 1) {
```

Comment:
BTW, Do we need this check as if scaleX=scaleY=1 then also no issue in doing multiplying with 1 ie scaleX/scaleY and if one of them is not 1 then we need to multiply anyway..so it seems this check is not needed?!!